implot: 0.16 -> 0.17

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/nix-community/nixpkgs-update) tools. This update was made based on information from https://repology.org/project/implot/versions.

meta.description for implot is: Immediate Mode Plotting

meta.homepage for implot is: https://github.com/epezent/implot




###### Updates performed
- Version update

###### To inspect upstream changes

- [Release on GitHub](https://github.com/epezent/implot/releases/tag/v0.17)

- [Compare changes on GitHub](https://github.com/epezent/implot/compare/v0.16...v0.17)

###### Impact

<b>Checks done</b>

---

- built on NixOS

> [!CAUTION]
> A test defined in `passthru.tests` did not pass.
- found 0.17 with grep in /nix/store/iybxhxvj052jagxjdiv1l63znhzqkql7-implot-0.17
- found 0.17 in filename of file in /nix/store/iybxhxvj052jagxjdiv1l63znhzqkql7-implot-0.17

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
2 total rebuild path(s)

2 package rebuild(s)

First fifty rebuilds by attrpath

implot
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from the cache**:
```
nix-store -r /nix/store/iybxhxvj052jagxjdiv1l63znhzqkql7-implot-0.17 \
  --option binary-caches 'https://cache.nixos.org/ https://nixpkgs-update-cache.nix-community.org/' \
  --option trusted-public-keys '
  nixpkgs-update-cache.nix-community.org-1:U8d6wiQecHUPJFSqHN9GSSmNkmdiFW7GW7WNAnHW0SM=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The nixpkgs-update cache is only trusted for this store-path realization.)
For the cached download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A implot https://github.com/r-ryantm/nixpkgs/archive/3fb55af70f486df746555c9a4d313be3bead6505.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/3fb55af70f486df746555c9a4d313be3bead6505#implot
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/iybxhxvj052jagxjdiv1l63znhzqkql7-implot-0.17
ls -la /nix/store/iybxhxvj052jagxjdiv1l63znhzqkql7-implot-0.17/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review --extra-nixpkgs-config '{ allowBroken = false; }'`
Commit: `3fb55af70f486df746555c9a4d313be3bead6505`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>implot</li>
  </ul>
</details>

---

###### Maintainer pings

cc @SomeoneSerge for [testing](https://github.com/nix-community/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs/blob/master/ci/README.md#nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-ryantm committed:
implot: 0.16 -> 0.17

wegank committed:
implot: fix tests

SuperSandro2000 commented:
This fails to apply the 0001-imgui-allow-installing-into-split-outputs.patch patch when rebase against master.

SomeoneSerge committed:
imgui: rebase cmakelists patch

Post vcpkg update (https://github.com/NixOS/nixpkgs/pull/507064),
and accommodating imnodes update (https://github.com/NixOS/nixpkgs/pull/505138#issuecomment-4210519258)
